### PR TITLE
Everything I Absolutely Will Not Discuss With Anyone Under Any Circumstances

### PR DESCRIPTION
**Pull request by TalkingCactus**
This PR will include the framework for the thing everyone does but doesn't talk about. The system will attempt to be as modular as possible (before I go insane) so others can incorporate their own things.

Built from the ground up without any of that garbage written by that garbage coder that was around a while ago or something.

Also rebuilds tgui for Pooj since I was having a few bugs that might be related to that. May or may not fix it. Should at least fix medihound sleeper ui.
Also adds:

- [x] Maintenance Bar
- [x] KinkMate
- [x] Door buttons are fixed, will open all linked doors at the same time
- [x] A bunch of bug fixes, re-balances, etc
- [x] Stuff I won't talk about 
- [ ] Overlays for junk
- [ ] Character preferences

### CODE REVIEW

File: code/datums/dna.dm [hunk 1]
```diff
@@ -247,6 +251,7 @@ mob/living/carbon/human/updateappearance(icon_update=1, mutcolor_update=0, mutat
 	eye_color = sanitize_hexcolor(getblock(structure, DNA_EYE_COLOR_BLOCK))
 	facial_hair_style = facial_hair_styles_list[deconstruct_block(getblock(structure, DNA_FACIAL_HAIR_STYLE_BLOCK), facial_hair_styles_list.len)]
 	hair_style = hair_styles_list[deconstruct_block(getblock(structure, DNA_HAIR_STYLE_BLOCK), hair_styles_list.len)]
+	hair_style = hair_styles_list[deconstruct_block(getblock(structure, DNA_HAIR_STYLE_BLOCK), hair_styles_list.len)]
```
Comment (ghost) [inline code/datums/dna.dm]: Did you accidentally paste this in a second time?

Comment (TalkingCactus) [reply code/datums/dna.dm]: That was a copypaste that I meant to change immediately. However I was probably distracted and committed it first.

**Comment by TalkingCactus** [comment]
DNA code is easy but confusing to read. 🔢

File: code/citadel/dildos.dm [hunk 2]
```diff
@@ -0,0 +1,82 @@
+//This is what happens when I get bored. Modular dildos is what happens.
```
Comment (ghost) [inline code/citadel/dildos.dm]: M O D U L A R  D I L D O S

**Comment by TalkingCactus** [comment]
Somehow we dodged conflicts. 🎉

**Comment by ghost** [comment]
status on this?

**Comment by TalkingCactus** [comment]
Framework done, polishing other things to work smoothly with it. Character creation implementation is underway currently.

File: code/__DEFINES/citadel_defines.dm [hunk 3]
```diff
@@ -0,0 +1,85 @@
+//Global defines for most of the unmentionables.
+//Be sure to update the min/max of these if you do change them.
+//Measurements are in imperial units. Inches, feet, yards, miles. Tsp, tbsp, cups, quarts, gallons. etc
+#define COCK_SIZE_BASE		3 //length in inches which is multiplied by the values below to estimate length and to generate a sprite for the mob
```
Comment (Iamgoofball) [inline code/__DEFINES/citadel_defines.dm:4]: alright time to review sex code

Comment (ghost) [reply code/__DEFINES/citadel_defines.dm:4]: http://i.imgur.com/kNGlrHo.png

**Comment by TalkingCactus** [comment]
someone can port this system to the new code if they want. I'll leave the branch alive for some time. I'm going to focus on fun things instead.

**Pull request closed without merging**